frontend/tests/utils/downloadUtils.ts

```diff
@@ -0,0 +1,387 @@
+    data[3].includes(`# Dataset Filter Values: ${datasetFilterValue}`)
+  ).toBeTruthy();
+  expect(
+    data[4].includes(`# Disease Filter Values: ${diseaseFilterValue}`)
+  ).toBeTruthy();
+  expect(
+    data[5].includes(
+      `# Self-Reported Ethnicity Filter Values: ${ethnicityFilterValue}`
+    )
+  ).toBeTruthy();
+  expect(
+    data[6].includes(`# Sex Filter Values: ${sexFilterValue}`)
+  ).toBeTruthy();
+};
+
+export async function downloadGeneFile(
+  page: Page,
+  tissues: string[],
+  subDirectory: string,
+  fileTypes: string[] = ["png"]
+): Promise<void> {
+  const allFileTypes = ["csv", "png", "svg"];
+  const dirPath = `${downLoadPath}/${subDirectory}`;
+
+  //wait for download file
+  const downloadPromise = page.waitForEvent("download");
+
+  //click the download icon
+  await page.getByTestId("download-button").click();
+  const CHECK = "Mui-checked";
+
+  for (const ext of allFileTypes) {
+    const checkboxId = `${ext}-checkbox`;
+    // uncheck if file type is checked but not
+    if (
+      (await page.getByTestId(checkboxId).getAttribute("class"))?.includes(
+        CHECK
+      ) &&
+      !fileTypes.includes(ext)
+    ) {
+      await page.getByTestId(checkboxId).locator("input").click();
+    }
+
+    // ensure wanted file types are checked
+    if (
+      !(await page.getByTestId(checkboxId).getAttribute("class"))?.includes(
+        CHECK
+      ) &&
+      fileTypes.includes(ext)
+    ) {
+      await page.getByTestId(checkboxId).locator("input").click();
+    }
+  }
+  await page.getByTestId("dialog-download-button").click();
+  const download = await downloadPromise;
+
+  // download can be zipped or not depending on number of tissues
+  let fileName = `${dirPath}/download.zip`;
+  if (fileTypes.length === 1 && tissues.length === 1) {
+    fileName = `${dirPath}/${tissues[0]}.${fileTypes[0]}`;
+  }
+
+  await download.saveAs(fileName);
+  //extract zip file
+  if (fileName.includes("zip")) {
+    const zip = new AdmZip(fileName);
+    zip.extractAllTo(dirPath);
+    console.log(dirPath);
+  }
+}
+
+export const getFilterText = async (page: Page, filterName: string) => {
+  const filter_label = `${getTestID(filterName)} [role="button"]`;
+  return await page.locator(filter_label).textContent();
+};
+export async function compareSvg(
+  page: Page,
+  webCellImage: string,
+  webGeneImage: string,
+  svgFile: string,
+  folder: string,
+  tissue: string
+): Promise<void> {
+  const svg = fs.readFileSync(`${downLoadPath}/${svgFile}`, "utf-8");
+  await page.setContent(svg);
+  // Wait for the page to finish loading
+  await page.waitForLoadState("networkidle");
+  const actualCell = `${downLoadPath}/${folder}/${tissue}1.png`;
+  //const actualGene = `${downLoadPath}/${folder}/${tissue}gene.png`;
+  await page
+    .locator("svg")
+    .locator("svg")
+    .nth(3)
+    .screenshot({ path: actualCell });
+  compareImages(actualCell, webCellImage);
+
+  //await page.screenshot()).toMatchSnapshot(webGeneImage);
```

Comment:
@tihuan there were issues with precision on the gene canvas; @victorikeokoro6  can elaborate more as he was the one who investigated and finally solved the issue

Reply:
toMatchSnapshot  is less flexible than pixelmatch because it relies on a pre-existing snapshot, which means it can miss new or subtle changes to the page that were not captured in the original snapshot.

Pixelmatch, on the other hand, is a standalone library that allows for more fine-grained control over the comparison process. With pixelmatch, you can specify the exact areas of the page that you want to compare, adjust the level of tolerance for differences between images.